admin_manual/installation/nginx.rst

```diff
@@ -114,75 +76,110 @@ webroot of your nginx installation. In this example it is
           fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
           fastcgi_param PATH_INFO $path_info;
           fastcgi_param HTTPS on;
-          # Avoid sending the security headers twice
-          fastcgi_param modHeadersAvailable true;
-          # Enable pretty urls
-          fastcgi_param front_controller_active true;
+          
+          fastcgi_param modHeadersAvailable true;         # Avoid sending the security headers twice
+          fastcgi_param front_controller_active true;     # Enable pretty urls
           fastcgi_pass php-handler;
+          
           fastcgi_intercept_errors on;
           fastcgi_request_buffering off;
       }
-
-      location ~ ^\/(?:updater|oc[ms]-provider)(?:$|\/) {
-          try_files $uri/ =404;
-          index index.php;
-      }
-
-      # Adding the cache control header for js, css and map files
-      # Make sure it is BELOW the PHP block
-      location ~ \.(?:css|js|woff2?|svg|gif|map)$ {
+      
+      location ~ \.(?:css|js|svg|gif)$ {
           try_files $uri /index.php$request_uri;
-          add_header Cache-Control "public, max-age=15778463";
-          # Add headers to serve security related headers (It is intended to
-          # have those duplicated to the ones above)
-          # Before enabling Strict-Transport-Security headers please read into
-          # this topic first.
-          #add_header Strict-Transport-Security "max-age=15768000; includeSubDomains; preload;" always;
-          #
-          # WARNING: Only add the preload option once you read about
-          # the consequences in https://hstspreload.org/. This option
-          # will add the domain to a hardcoded list that is shipped
-          # in all major browsers and getting removed from this list
-          # could take several months.
-          add_header Referrer-Policy "no-referrer" always;
-          add_header X-Content-Type-Options "nosniff" always;
-          add_header X-Download-Options "noopen" always;
-          add_header X-Frame-Options "SAMEORIGIN" always;
-          add_header X-Permitted-Cross-Domain-Policies "none" always;
-          add_header X-Robots-Tag "none" always;
-          add_header X-XSS-Protection "1; mode=block" always;
-
-          # Optional: Don't log access to assets
-          access_log off;
+          expires 6M;         # Cache-Control policy borrowed from `.htaccess`
+          access_log off;     # Optional: Don't log access to assets
       }
-
-      location ~ \.(?:png|html|ttf|ico|jpg|jpeg|bcmap|mp4|webm|mp3|ogg|wav)$ {
+      
+      location ~ \.woff2?$ {
           try_files $uri /index.php$request_uri;
-          # Optional: Don't log access to other assets
-          access_log off;
+          expires 7d;         # Cache-Control policy borrowed from `.htaccess`
+          access_log off;     # Optional: Don't log access to assets
+      }
+      
+      location / {
+          try_files $uri $uri/ /index.php$request_uri;
```

Comment:
Why does this need its own location, can it just live above or below the `index` parameter?

Reply:
After some googling, I found that it doesn't, but the behaviour of `try_files` in this regard doesn't appear to be documented, and in my experience, it seems to be standard practice to not use `try_files` in `server` scope if there are `location` blocks. IMO, it eases readability and avoids confusion. Even if we take it out of the `location` block, I personally would leave it after the last location block, not place it directly after `index`.

Reply:
Ok, the main reason I  think it would be good to not nest this fallback try_files in a location is because the `location = /` we made earlier wont inherit this. But we could easily fix this by nesting that `location =` in this fallback location.

Reply:
A `try_files` directive in `server` scope is not inherited by any child `location` blocks, anyway. Instead, it is used iff the URI does not match any `location` blocks. We don't need a `try_files` within `location = /`, because the `index` directive results in `/index.php` being served (if the user agent doesn't match `^DavClnt`).

The following are all behaviourally equivalent:

```nginx
server {
    index index.php index.html /index.php$request_uri;

    location = / {
        if ( $http_user_agent ~ ^DavClnt ) {
            return 302 /remote.php/webdav/$is_args$args;
        }
    }

    location / {
        try_files $uri $uri/ /index.php$request_uri;
    }
}
```

```nginx
server {
    index index.php index.html /index.php$request_uri;

    location = / {
        if ( $http_user_agent ~ ^DavClnt ) {
            return 302 /remote.php/webdav/$is_args$args;
        }
    }

    try_files $uri $uri/ /index.php$request_uri;
}
```

```nginx
server {
    index index.php index.html /index.php$request_uri;

    try_files $uri $uri/ /index.php$request_uri;
    
    location = / {
        if ( $http_user_agent ~ ^DavClnt ) {
            return 302 /remote.php/webdav/$is_args$args;
        }
    }
}
```

Reply:
Good point! I did not think to realize `try_files` is irrelevant with `location =`